src/cosmoz-bottom-bar.ts

```diff
@@ -266,23 +189,73 @@ const CosmozBottomBar = ({ active = false, maxToolbarItems = 1 }: Props) => {
 		[active],
 	);
 
+	const calculateDistribution = () => {
```

Comment:
You should memoize this so you compute the distribution only when the states change.

```suggestion
	const distribution = useMemo(() => {...}, [buttonStates]);
```